### PR TITLE
Integration test cleanup for cloudscale modules

### PR DESCRIPTION
**Pull request by gaudenz**
##### SUMMARY
This refactors the cleanup procedure for the integration tests of the
cloudscale_server and cloudscale_volume modules to use an "always"
section for cleanup. The cleanup code also deletes all resources which
contain the test run prefix. This ensures that all resources are cleaned
up regardless of the actual test result which is a prerequisite for
running these tests in CI.

##### ISSUE TYPE
- Test improvement Pull Request

##### COMPONENT NAME
cloudscale_server
cloudscale_volume

### CODE REVIEW

File: test/integration/targets/cloudscale_floating_ip/aliases [hunk 1]
```diff
@@ -0,0 +1,2 @@
+cloud/cloudscale
+unsupported
```
Comment (mattclay) [inline test/integration/targets/cloudscale_floating_ip/aliases:2]: What is preventing these tests from being run in CI?

Comment (mattclay) [reply test/integration/targets/cloudscale_floating_ip/aliases:2]: Ah, never mind. I read this and thought cloudstack instead of cloudscale.

Comment (gaudenz) [reply test/integration/targets/cloudscale_floating_ip/aliases:2]: We are working on it ;-) This is basically a prerequisite for this as it ensures that tests cleanup reasources regardless of success.

**Review by resmo — APPROVED** [review]
LGTM

**Pull request merged**